staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go

```diff
@@ -122,6 +123,7 @@ func (o *WardleServerOptions) Config() (*apiserver.Config, error) {
 		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
 	}
 
+	o.RecommendedOptions.Admission.DisablePlugins = []string{validatingadmissionpolicy.PluginName}
```

Comment:
weird.  Why?

Reply:
This is for fixing the test for aggregated APIserver in https://github.com/kubernetes/kubernetes/blob/9043ce05c125091c0cb5519206fd90d311abd8c8/test/integration/examples/apiserver_test.go#L231.
In this test, it will try to start an aggregated test server from here. However, after we added rest mapper into plugin Initialization, it makes it hard to pass in a plugin initializer with rest mapper here due to the dependency issues(staging cannot depend on k/k in tree). Since the test is nothing to do with plugins. I disabled it from here instead of trying to move dependency around.

Reply:
After further research, I would go with removing VAP plugin from https://github.com/kubernetes/kubernetes/blob/9043ce05c125091c0cb5519206fd90d311abd8c8/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L89 since I don't think VAP should fit there. Let me update with a separate commit for review.

Reply:
Update: I have reverted the changes here. And fix the test within the test case itself. Thanks

Reply:
> However, after we added rest mapper into plugin Initialization, it makes it hard to pass in a plugin initializer with rest mapper here due to the dependency issues(staging cannot depend on k/k in tree).

I'm not sure I understand this ... RESTMappers interfaces and implementations live in apimachinery, which all staging repos can depend on.

If generic admission plugins that live in k8s.io/apiserver require a RESTMapper to function, shouldn't the RESTMapper plumbing to admission be moved from pkg/kubeapiserver/admission/initializer.go#NewPluginInitializer to k8s.io/apiserver/pkg/admission/initializer/initializer.go#New?

Reply:
Thank you! I have moved restmapper to staging in a latest commit. Thanks!